pkg/sql/importer/import_job.go

```diff
@@ -1097,6 +1101,35 @@ func (r *importResumer) publishSchemas(ctx context.Context, execCfg *sql.Executo
 	})
 }
 
+// checkVirtualConstraints checks constraints that are enforced via runtime
+// checks, such as uniqueness checks that are not directly backed by an index.
+func (*importResumer) checkVirtualConstraints(
+	ctx context.Context, execCfg *sql.ExecutorConfig, job *jobs.Job,
+) error {
+	for _, tbl := range job.Details().(jobspb.ImportDetails).Tables {
+		desc := tabledesc.NewBuilder(tbl.Desc).BuildExistingMutableTable()
+		desc.SetPublic()
+
+		if sql.HasVirtualUniqueConstraints(desc) {
+			if err := job.RunningStatus(ctx, nil /* txn */, func(_ context.Context, _ jobspb.Details) (jobs.RunningStatus, error) {
+				return jobs.RunningStatus(fmt.Sprintf("re-validating %s", desc.GetName())), nil
+			}); err != nil {
+				return errors.Wrapf(err, "failed to update running status of job %d", errors.Safe(job.ID()))
```

Comment:
Doesn't matter, but just as an FYI, jobspb.JobID implements SafeValue so it should work without wrapping it in errors.Safe.